recipe/build.sh

```diff
@@ -35,5 +35,6 @@ ctest \
 |^cxx.test_scope_crash$\
 |^cxx.test_std_size_t$\
 |^cxx.test_stl_container_provides_allocator$\
+|^cxx.test_typeid$\
```

Comment:
Skipping one more test based on discussion in https://github.com/include-what-you-use/include-what-you-use/issues/1033#issuecomment-1116260796 and https://github.com/conda-forge/staged-recipes/pull/18539#discussion_r863321869

It would be great to fix this but I don't really have the expertise.